src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -709,4 +827,32 @@ public static String swapToolGrade(final int toolGrade)
                 return "";
         }
     }
+
+    /**
+     * Adapted from {@link net.minecraft.entity.player.InventoryPlayer#addItemStackToInventory(ItemStack)}.
+     *
+     * @param inventory Inventory to add itemstack to.
+     * @param itemStack ItemStack to add.
+     * @param building the building.
+     * @return itemStack which has been replaced.
+     */
+    @Nullable
+    public static ItemStack forceItemStackToInventory(@NotNull final IInventory inventory, @NotNull final ItemStack itemStack, @NotNull final AbstractBuilding building)
+    {
+        if(!addItemStackToInventory(inventory, itemStack))
+        {
+            final List<ItemStorage> localAlreadyKept = new ArrayList<>();
+            for(int i = 0; i < inventory.getSizeInventory(); i++)
+            {
+                final ItemStack localStack = inventory.getStackInSlot(i);
+                if(!EntityAIWorkDeliveryman.workerRequiresItem(building, localStack, localAlreadyKept))
+                {
+                    final ItemStack removedStack = inventory.removeStackFromSlot(i);
+                    inventory.setInventorySlotContents(i, itemStack.copy());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'itemStack' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2259)